src/c++/perf_analyzer/genai-pa/genai_pa/llm_metrics.py

```diff
@@ -43,29 +43,52 @@
     from transformers import AutoTokenizer
 
 
-@dataclass
-class LLMMetrics:
+class Metrics:
+    """A base class for all the metrics class that contains common metrics."""
+
+    def __init__(
+        self,
+        request_throughputs: list[float] = [],
+        request_latencies: list[int] = [],
+    ) -> None:
+        self.request_throughputs = request_throughputs
+        self.request_latencies = request_latencies
+        self._base_names = {
+            "request_throughputs": "request_throughput",
+            "request_latencies": "request_latency",
+        }
+
+    @property
+    def data(self) -> dict:
+        return {k: v for k, v in self.__dict__.items() if not k.startswith("_")}
+
+    def get_base_name(self, metric_name: str) -> str:
+        if metric_name in self._base_names:
+            return self._base_names[metric_name]
+        else:
+            raise KeyError(f"No metric named '{metric_name}' exists.")
+
+
+class LLMMetrics(Metrics):
     """A simple dataclass that holds core LLM performance metrics."""
 
-    time_to_first_tokens: list[int]
-    inter_token_latencies: list[int]
-    output_token_throughputs: list[int]
-
-    def get_base_name(self, attr_name: str) -> str:
-        # Attempted to extract and store the mapping as a dataclass member as a
-        # dictionary but encountered two issues: (1) Python does not allow
-        # dataclass member to be mutable and (2) if we set it as member of
-        # normal class, the dict member will be parsed by Statistics class,
-        # which is not what we want since it's not one of the LLM metrics.
-        # Leaving it as conditional statements for now.
-        if attr_name == "time_to_first_tokens":
-            return "time_to_first_token"
-        elif attr_name == "inter_token_latencies":
-            return "inter_token_latency"
-        elif attr_name == "output_token_throughputs":
-            return "output_token_throughput"
-        else:
-            raise ValueError(f"No attribute named '{attr_name}' exists.")
+    def __init__(
+        self,
+        request_throughputs: list[float] = [],
+        request_latencies: list[int] = [],
+        time_to_first_tokens: list[int] = [],
+        inter_token_latencies: list[float] = [],
```

Comment:
Great work on this ticket. Some really nice refactoring!

Why are inter-token latencies being changed to a float? Are all of the time units still nanoseconds?

Reply:
They are all still nanoseconds since all the timestamps from PA are in nanoseconds and I don't do any conversions. The ITL changed to float because the calculation changed: we divide the difference between the two timestamps by the number of tokens generated by the latter one.

Reply:
Can we round the result? If we're expecting ITL to be <10ns, then it'd make sense to float. If we're not, then adding decimal places obscures what the user wants to know.

Pretty_print already cuts off the decimal places, which is why it was important for me to know whether we are adding a metric where decimal places matter. But I think we generally don't need float precision for metrics with nanosecond units.

Reply:
Sure I can round the results. Don't think it will affect the stats much.

Reply:
Thanks!